template/CHART_NAME/templates/daemonset.yaml

```diff
@@ -0,0 +1,126 @@
+      {{- end }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.podSecurityContext.enabled }}
+      securityContext:
+        fsGroup: {{ .Values.%%MAIN_OBJECT_BLOCK%%.podSecurityContext.fsGroup }}
+      {{- end }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.initContainers }}
+      initContainers: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.initContainers "context" $) | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: %%CONTAINER_NAME%%
+          image: {{ template "%%TEMPLATE_NAME%%.image" . }}
+          imagePullPolicy: {{ .Values.%%MAIN_OBJECT_BLOCK%%.image.pullPolicy }}
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.containerSecurityContext.enabled }}
+          securityContext:
+            runAsUser: {{ .Values.%%MAIN_OBJECT_BLOCK%%.containerSecurityContext.runAsUser }}
+            runAsNonRoot: {{ .Values.%%MAIN_OBJECT_BLOCK%%.containerSecurityContext.runAsNonRoot }}
+          {{- end }}
+          command:
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.command }}
+          {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.command "context" $) | nindent 12 }}
+          {{- else }}
+            %%CONTAINER_COMMAND%%
+          {{- end }}
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.args "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            - name: foo
+              value: bar
+            {{- if .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVars }}
+            {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVars "context" $) | nindent 12 }}
+            {{- end }}
+          envFrom:
+            {{- if .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVarsCM }}
+            - configMapRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVarsCM "context" $) }}
+            {{- end }}
+            {{- if .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVarsSecret }}
+            - secretRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.extraEnvVarsSecret "context" $) }}
+            {{- end }}
+          ports:
+            - containerPort: foo
+              protocol: bar
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.resources }}
+          resources: {{- toYaml .Values.%%MAIN_OBJECT_BLOCK%%.resources | nindent 12 }}
+          {{- end }}
+          {{- if .Values.%%MAIN_OBJECT_BLOCK%%.livenessProbe.enabled }}
+          livenessProbe: {{- omit .Values.%%MAIN_OBJECT_BLOCK%%.livenessProbe "enabled" | toYaml | nindent 12 }}
```

Comment:
Don't use `omit` + `toYaml` this in this case. Unlike securityContext, this could lead to incompatible probes as it could merge the default values with extra probes.